### ISSUE TITLE
Pagetual: Site Acts Weird

### ISSUE THREAD

**Issue by rahaaatul** [hoothin/UserScripts#139]
# Description
When pagetual script is on, this site acts weird. See screenshot below.

# Steps to reproduce the behavior
1. Go to [Mobilism](https://forum.mobilism.org/search.php/) 
2. Search something
4. See error.

# Expected behavior
![Normal](https://user-images.githubusercontent.com/86149861/181443564-4fb71414-cc3b-49dd-93dd-3fa2f60576bb.jpg)

# Actual behavior
![Abnormal](https://user-images.githubusercontent.com/86149861/181443585-f6a42f19-f95a-435e-bc10-2dcddefc6df2.jpg)

# Device Information
 - Device: Vivo V2043
 - OS: Android 11
 - Browser: Kiwi Browser (103.0.5060.104)
 - Pagetual Version: 1.9.30.3.11

**Comment by hoothin** [comment]
<pre name="pagetual">
    {
        "name": "Mobilism",
        "type": 0,
        "url": "^https://forum\\.mobilism\\.org/",
        "pageElement": "id('page-body')//section|id('page-body')/main/table[last()]/tbody/tr|//div[@class='row-fluid blue-bar']",
        "nextLink": "//a[i/@class='icon-chevron-right']",
        "css": "td.center { display: none; }"
    }
</pre>

**Comment by rahaaatul** [comment]
@hoothin I thought it was happening only on search results but it's happening all over the site, for example: https://forum.mobilism.org/viewforum.php?f=399
![Screenshot_2022_0728_171737](https://user-images.githubusercontent.com/86149861/181492847-c39eaa8f-06fe-4dd6-9b11-7e7d9bfa6600.jpg)

**Comment by hoothin** [comment]
<pre name="pagetual">
    {
        "name": "Mobilism",
        "type": 0,
        "url": "^https://forum\\.mobilism\\.org/",
        "pageElement": "id('page-body')//section|id('page-body')/main/table[last()]/tbody/tr|//div[@class='row-fluid blue-bar']",
        "nextLink": "//a[i/@class='icon-chevron-right']",
        "css": ".stats-color,.last-color,td.center { display: none; }"
    }
</pre>

**Comment by hoothin** [comment]
That site is really weird, it has the different structure between desktop & mobile on the same url.
So the rule I pasted before will effect only on mobile phone.
You can made one by yourself if you know a little css.

**Comment by rahaaatul** [comment]
> {
>         "name": "Mobilism",
>         "type": 0,
>         "url": "^https://forum\\.mobilism\\.org/",
>         "pageElement": "id('page-body')//section|id('page-body')/main/table[last()]/tbody/tr|//div[@class='row-fluid blue-bar']",
>         "nextLink": "//a[i/@class='icon-chevron-right']",
>         "css": ".stats-color,.last-color,td.center { display: none; }"
>     }

It's fixed, now. Thank you, @hoothin .

**Comment by rahaaatul** [comment]
> That site is really weird, it has the different structure between desktop & mobile on the same url. So the rule I pasted before will effect only on mobile phone. You can made one by yourself if you know a little css.

I don't know CSS, but I'd love to learn it.

**Comment by rahaaatul** [comment]
@hoothin

https://www.linuxfromscratch.org/blfs/view/svn/postlfs/nano.html

![IMG_20220804_145717.jpg](https://user-images.githubusercontent.com/86149861/182807035-fd1f8916-6030-4362-a645-6db473110878.jpg)

**Comment by hoothin** [comment]
<pre name="pagetual">
    {
        "name": "linuxfromscratch",
        "url": "^https://www\\.linuxfromscratch\\.org",
        "pageElement": "body>*",
        "css": ".sect1{overflow-wrap: break-word;}.navheader>ul{display:none;}"
    }
</pre>

**Comment by rahaaatul** [comment]
@hoothin
https://forum.xda-developers.com/t/binary-gnu-nano-editor-v2-2-6-for-android.1530519/
![chrome_screenshot_Aug 4, 2022 6_16_13 PM GMT+06_00.png](https://user-images.githubusercontent.com/86149861/182844636-66eade58-1939-4249-9149-ee22a1947df0.png)

**Comment by hoothin** [comment]
Pagetual did not run on this page, so that style may be cause by any other extension.

**Comment by rahaaatul** [comment]
@hoothin 
This site doesn't show thumbnails for below contents and keeps loading but pages never load.

https://user-images.githubusercontent.com/86149861/183237438-16a8e50e-3ef9-4304-b091-d34e32000878.mp4

**Comment by hoothin** [comment]
Fixed. Maybe you can paste the URL here next time.
<pre name="pagetual">
    {
        "name": "Linux Tips, Tricks and Tutorials | Linuxize",
        "url": "^https://linuxize\\.com",
        "pageElement": ".post-list>.flex-wrap>section",
        "lazyImgSrc": "data-cfsrc"
    }
</pre>

**Comment by rahaaatul** [comment]
> Fixed. Maybe you can paste the URL here next time.
> <pre name="pagetual">
>     {
>         "name": "Linux Tips, Tricks and Tutorials | Linuxize",
>         "url": "^https://linuxize\\.com",
>         "pageElement": ".post-list>.flex-wrap>section",
>         "lazyImgSrc": "data-cfsrc"
>     }
> </pre>

Sorry, my bad! I forgot to do that.

**Comment by hoothin** [comment]
You need update the script first

**Comment by rahaaatul** [comment]
> You need update the script first

Everytime I comment now, I first reinstall/upgrade and update the script.

As I said, this occurs without pagetual or any extensions on. Tried on total vanilla browser, thumbnails doesn't load.

**Comment by hoothin** [comment]
![Screenshot_20220806_234638](https://user-images.githubusercontent.com/5361213/183256186-af88fe0f-78fb-45d1-85d5-ba2f9d1e365f.jpg)
I tried on kiwi and it works.

**Comment by rahaaatul** [comment]
> ![Screenshot_20220806_234638](https://user-images.githubusercontent.com/5361213/183256186-af88fe0f-78fb-45d1-85d5-ba2f9d1e365f.jpg)
> I tried on kiwi and it works.

I'm not talking about these thumbnails, when you scroll down in every page, you will see a pattern:

Each page loads 4 thumbnails for first 4 links, after that there are no thumbnails in later linka. And it's on every page.

That's what I was talking about.

**Comment by rahaaatul** [comment]
https://trakt.tv/shows/stranger-things/seasons/4/episodes/1

Doesn't matter really, but these are missing from next page.
- Actor's Name and Photo.
- Watching Now.
- Comments.
![chrome_screenshot_Aug 7, 2022 12_08_28 AM GMT+06_00.png](https://user-images.githubusercontent.com/86149861/183260988-a98f689a-c091-4868-ab06-f56d28256efa.png)

**Comment by hoothin** [comment]
<pre name="pagetual">
    {
        "name": "Trakt",
        "url": "^https://trakt\\.tv",
        "pageElement": "section#info-wrapper > div.container > div.row > div#overview>*",
        "action": 1,
        "waitElement": [
            ".summary-activity",
            "#popular-comments.fade"
        ]
    }
</pre>
So you are also a fan of Stranger-Things!

**Comment by rahaaatul** [comment]
> So you are also a fan of Stranger-Things!
Yes! Watched all of them, it was so good! Especially season 4, can't wait for season 5. Who is your favorite character from the show? Mine is Dusty-bun 😂.

**Comment by rahaaatul** [comment]
> So you are also a fan of Stranger-Things!

Yes! Watched all of them, it was so good! Especially season 4, can't wait for season 5. Who is your favorite character from the show? Mine is Dusty-bun 😂.

**Comment by rahaaatul** [comment]
https://1337x.to/search/something/1/

Not working on 1337x.to.

**Comment by hoothin** [comment]
Absolutely be little El😄.
I checked 1337x.to under kiwi and it looks like acting well. Can you check again?

**Comment by rahaaatul** [comment]
> Absolutely be little El😄.
She's scary too.

> I checked 1337x.to under kiwi and it looks like acting well. Can you check again?
I did. And, it's working now. Weird. 😕

**Comment by rahaaatul** [comment]
[Soccercatch](https://www.soccercatch.com/leagues/8/premier-league#shows)

Doesn't auto stitch, basically all over this website, same problem.

**Comment by hoothin** [comment]
<pre name="pagetual">
    {
        "name": "Soccercatch.com",
        "url": "^https://www\\.soccercatch\\.com.*#shows",
        "nextLink": ".tabs-page[data-title=shows] .pagination-next",
        "init": "if(doc)doc.querySelector('[data-title=shows]').click();",
        "pageElement": ".show-item"
    }
</pre>
This rule is for shows tab, you can modify for other tabs.

**Comment by rahaaatul** [comment]
https://www.16personalities.com/free-personality-test

Tried disabling for this site but didn't work. Checking choices in next pages, doesn't work. Can you check?

**Comment by hoothin** [comment]
So you means that you can't disable for this site?
I checked the site, there is code about recording & calculating your choices.
I disabled pagetual and it works well. What's your trouble?

**Comment by rahaaatul** [comment]
> So you means that you can't disable for this site?

I disabled, and it showed it disabled. I refreshed and I tapped next page choices, but they were not selected.

> I disabled pagetual and it works well. What's your trouble?

It works for me now, too. Possibly problem with browser. I restarted it and it worked.

Sorry for bothering you.

**Comment by rahaaatul** [comment]
> <pre name="pagetual">
>     {
>         "name": "Soccercatch.com",
>         "url": "^https://www\\.soccercatch\\.com.*#shows",
>         "nextLink": ".tabs-page[data-title=shows] .pagination-next",
>         "init": "if(doc)doc.querySelector('[data-title=shows]').click();",
>         "pageElement": ".show-item"
>     }
> </pre>
> This rule is for shows tab, you can modify for other tabs.

This one isn't working, can you check again?

**Comment by hoothin** [comment]
Ah, I tested only on desktop, I'll check mobile for you soon.

**Comment by rahaaatul** [comment]
https://www.gettyimages.ie/photos/manchester-united-tottenham?family=editorial&assettype=image&phrase=hoothin&sort=mostpopular

Doesn't stitch next pages.

**Comment by hoothin** [comment]
Take it.
<pre name="pagetual">
    {
        "name": "Manchester Pictures and Photos - Getty Images",
        "url": "^https://www\\.gettyimages\\.ie",
        "nextLink": "button[title='Next page']"
    }
</pre>

**Comment by rahaaatul** [comment]
> Take it.
> <pre name="pagetual">
>     {
>         "name": "Manchester Pictures and Photos - Getty Images",
>         "url": "^https://www\\.gettyimages\\.ie",
>         "nextLink": "button[title='Next page']"
>     }
> </pre>

Not working. Can you check in kiwi?

Also change its name to **Search Getty Images**.

**Comment by hoothin** [comment]
<pre name="pagetual">
    {
        "name": "Search Getty Images",
        "url": "^https://www\\.gettyimages\\.ie",
        "nextLink": "button[title='Next page']",
        "pageElement": "[data-testid=galleryMosaicAsset]"
    }
</pre>
Use this in kiwi.

**Comment by rahaaatul** [comment]
@hoothin long time no see! How are you doing? Here's one defective site:
https://www.daraz.com.bd/vacuum-cleaners/

**Comment by hoothin** [comment]
Thanks for report. Fixed now.

**Comment by rahaaatul** [comment]
Mobilism isn't auto paging...
https://forum.mobilism.org/search.php?author_id=636409&sr=posts

Can you fix this, please? @hoothin

**Comment by rahaaatul** [comment]
Hey @hoothin , please, fix this one... Footer appears in the middle of the spliced pages.

![Screenshot (6)](https://github.com/hoothin/UserScripts/assets/86149861/94591f10-cbcc-4857-9c2d-900719ec1c30)

**Comment by hoothin** [comment]
<pre name="pagetual">
{
    "name": "Search | Mobilism",
    "url": "^https?://forum\\.mobilism\\.org/search\\.php\\?",
    "pageElement": "main > div.row-fluid > div.well.well-small"
}
</pre>

**Comment by rahaaatul** [comment]
When scrolling, [this site](https://yts.mx/browse-movies) acts weird. I scroll down, it scrolls up automatically.


https://github.com/hoothin/UserScripts/assets/86149861/abc72487-2091-4085-9ba4-daaefe59f363

**Comment by hoothin** [comment]
Have you checked "contentVisibility"? If so, uncheck it.

**Comment by rahaaatul** [comment]
Can you check [this site](https://www.waltdisneystudios.jp/ghibli)? I tried this but does not work:

```css
[
    {
        "name": "Ghibli",
        "url": "^https://www\\.waltdisneystudios\\.jp/",
        "pageElement": "#ref-1-2 > div.bound.movie > div.show_more_container > button.show_more.blue.button.large.ada-el-focus"
    }
]

**Comment by rahaaatul** [comment]
@hoothin how to hide this:
![Screenshot_2023_0720_231212](https://github.com/hoothin/UserScripts/assets/86149861/c175c57f-790e-4d54-97ec-4fc1018762f5)

This is the [site](https://mydramalist.com/search?adv=titles&ty=68,83&so=rated).

**Comment by rahaaatul** [comment]
> {
>         "name": "Mobilism",
>         "type": 0,
>         "url": "^https://forum\\.mobilism\\.org/",
>         "pageElement": "id('page-body')//section|id('page-body')/main/table[last()]/tbody/tr|//div[@class='row-fluid blue-bar']",
>         "nextLink": "//a[i/@class='icon-chevron-right']",
>         "css": ".stats-color,.last-color,td.center { display: none; }"
>     }
> Import

This is no longer working @hoothin , can you check this again, please?

**Comment by rahaaatul** [comment]
@hoothin can you fix the splitting for this site?

https://www.pexels.com/@stefwithanf-1955763/

![image](https://github.com/hoothin/UserScripts/assets/86149861/b68ef56e-e7a9-405e-91fb-3b8a627398af)
